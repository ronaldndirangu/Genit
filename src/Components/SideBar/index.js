import React from "react";
import Input from "../Input";
import Button from "../Button";
import "./index.scss";

const Sidebar = ({ handleChange, handleClick, amount, disable }) => {
  return (
    <div className="sidebar">
      <div className="sidebar-text">
        Generate upto 10,000 random phone numbers
      </div>
      <Input action={handleChange} value={amount || ""} />
      <Button action={handleClick} disable={disable} />
    </div>
  );
};

export default Sidebar;
