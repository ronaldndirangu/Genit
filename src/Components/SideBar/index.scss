.sidebar {
  display: flex;
  position: relative;
  flex-direction: column;
  height: calc(100vh - 6rem);
  width: 18rem;
  border: solid 1px #ccc;
  padding: 1rem;

  &-text {
    display: flex;
    justify-content: center;
    text-align: center;
    margin-bottom: 2rem;
    color: #39298c;
    font-weight: 400;
  }
}
