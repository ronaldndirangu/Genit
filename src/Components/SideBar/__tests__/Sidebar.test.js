import React from "react";
import { shallow } from "enzyme";
import SideBar from "../../SideBar";

describe("SideBar", () => {
  it("SideBar renders correctly", () => {
    const component = shallow(<SideBar />);
    expect(component).toMatchSnapshot();
  });
});
