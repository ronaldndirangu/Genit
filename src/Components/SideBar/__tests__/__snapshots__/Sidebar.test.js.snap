// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar SideBar renders correctly 1`] = `ShallowWrapper {}`;
